src/builtin-addons/core/template-completion-provider.ts

```diff
@@ -62,6 +64,59 @@ const mListHelpers = memoize(listHelpers, { length: 1, maxAge: 60000 }); // 1 se
 
 const mListRoutes = memoize(listRoutes, { length: 1, maxAge: 60000 });
 
+/**
+ * Generates a map of completion label (file name) to array of potential namespaced
+ * paths.
+ * @param addonsMeta addons meta array
+ * @param server Server
+ * @param focusPath currentfocus path
+ * @returns { [key: string]: string[] }
+ */
+function getResultsObj(addonsMeta: Array<AddonMeta>, server: Server, focusPath: ASTPath) {
```

Comment:
could we rename it to get more context for developers, like "generateNamespacedComponentsHashMap" or such..
also, have basic unit test for it will be great!

Reply:
done :)